_posts/2020-04-07-hhvm-4.52.markdown

```diff
@@ -0,0 +1,28 @@
+---
+title: "HHVM 4.52"
+layout: post
+author: fred
+category: blog
+---
+
+HHVM 4.52 is released! This release marks the end of support for 4.46;
+HHVM 4.47&ndash;4.51 remain supported, as do the 4.8 and 4.32 LTS releases.
+
+# Highlights
+
+- fixed crash in `ext_soap` if a SoapServer's handler class does not exist.
+- `?>` and `<?=` are no longer recognized by the parser; this has replaced
+  dedicated errors with parse errors.
+
+# Breaking Changes
+
+- `__halt_compiler()` has been removed.
+- it is now a type error to `use` the same trait twice in the same class.
```

Comment:
Does this include via inheritance?
E.g. `BaseClass` uses `SomeTrait` and `ChildClass` uses `SomeTrait` and `ChildClass` extends `BaseClass`.

Reply:
nope, will expand

Reply:
It does not, but although there is a typechecker option `no_trait_reuse` to forbid this. That might become the default at some point.